File: types/exam.ts
export interface Exam {
  id: number
  subject: string
  code: string
  date: string
  time: string
  duration_minutes: number
  duration_day?: string // P1D, P2D, etc. for multi-day exams
  location: string
  comment?: string
  school: string
  degree: string
  year: string
  semester: string
  acronym?: string
  created_at?: string
}

export interface ExamFilters {
  school?: string | string[]
  degree?: string | string[]
  year?: string
  semester?: string
  subject?: string
  acronym?: string
}

File: backend/src/main/java/com/upv/examcalendar/model/Exam.java
package com.upv.examcalendar.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "ETSINF")
public class Exam {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "subject")
    private String subject;

    @Column(name = "degree")
    private String degree;

    @Column(name = "year")
    private String year;

    @Column(name = "semester")
    private String semester;

    @Column(name = "exam_date")
    private LocalDateTime date;

    @Column(name = "place")
    private String room;

    @Column(name = "school")
    private String school;

    // Additional fields that might exist in the ETSINF table
    @Column(name = "exam_instance_id")
    private String examInstanceId;

    @Column(name = "exam_time")
    private String examTime;

    @Column(name = "code")
    private String code;

    @Column(name = "acronym")
    private String acronym;

    @Column(name = "duration_minutes")
    private Integer durationMinutes;

    @Column(name = "comment")
    private String comment;
}